client/src/app/site/motions/components/motion-import-list/motion-import-list.component.ts

```diff
@@ -0,0 +1,265 @@
+    }
+
+    public get nonImportableCount(): number {
+        if (this.importer && this.hasFile) {
+            return this.importer.summary.errors + this.importer.summary.duplicates;
+        }
+        return null;
+    }
+
+    public get doneCount(): number {
+        return (this.importer && this.hasFile) ? this.importer.summary.done : null;
+    }
+
+    /**
+     * Constructor for list view bases
+     * @param titleService the title serivce
+     * @param translate the translate service
+     * @param matSnackBar
+     * @param importer: The user csv import service
+     */
+    public constructor( titleService: Title, public translate: TranslateService,
+        matSnackBar: MatSnackBar, private importer: MotionImportService,
+        private motionCSVExport: MotionCsvExportService) {
+        super(titleService, translate, matSnackBar);
+        this.initTable();
+        this.selectedEncoding = this.encodings[0];
+        this.importer.errorEvent.subscribe( message => this.raiseError(message) );
+    }
+
+    /**
+     * Children need to call this in their init-function.
+     * Calling these three functions in the constructor of this class
+     * would be too early, resulting in non-paginated tables
+     */
+    public initTable(): void {
+        this.dataSource = new MatTableDataSource();
+        const me = this;
+        this.setFilter();
+        this.importer.getNewEntries().subscribe(newEntries => {
+            me.dataSource.data = newEntries;
+            if (newEntries.length){
+                me.hasFile = true;
+            }
+        });
+    }
+
+    /**
+     * Returns the table column definition
+     */
+    public getColumnDefinition(): string[]{
+        return ['status'].concat(this.importer.expectedHeader);
+    }
+
+    /**
+     * toggle for 'done' checkbox
+     * TODO: more angular-material way to do this
+     */
+    public toggleHiddenDone() : void {
+        this.hidden.done = !this.hidden.done;
+        this.setFilter();
+    }
+
+    /**
+     * toggle for 'errors' checkbox
+     * TODO: more angular-material way to do this
+     */
+    public toggleHiddenErrors() : void {
+        this.hidden.errors = !this.hidden.errors;
+        this.setFilter();
+    }
+
+    /**
+     * triggers the importer's onSelectFile
+     */
+    public onSelectFile(event: any): void {
+        this.importer.onSelectFile(event, this.selectedEncoding);
+    }
+
+    /**
+     * Triggers the importer's import
+     */
+    public async doImport(): Promise<void> {
+        await this.importer.doImport();
+        this.setFilter();
+    }
+
+    /**
+     * Updates and manually triggers the filter function.
+     * See {@link hidden} for options
+     * (changed from default mat-table filter)
+     */
+    private setFilter(): void {
+        const me = this;
+        this.dataSource.filter = '';
+        this.dataSource.filterPredicate = function (
```

Comment:
Please use an arrow function. This has the advantage, that you do not need the `me`, becuase an arrow function does not have it's own scope.